### DISCUSSION TITLE
BottomSheet closes and then reopens immediately

### DISCUSSION THREAD

**Discussion by lintonye** [gorhom/react-native-bottom-sheet#1160] [Q&A]
Hi,

It seems the BottomSheet fails to close when `close` is invoked inside a callback.

This code fails:
```ts
  // createPoolMutation is a React Query mutation created with `useMutation`
  const onSubmit = (data) => {
    createPoolMutation.mutate(data, {
      onSuccess: (response) => {
        console.log(response);
        bottomSheetRef.current.close();
      },
    });
  };
```

**However**, the strange thing is that the code below works in the same codebase (I just commented out the `createPoolMutation` part and added a call to `close`):

```ts
  const onSubmit = (data) => {
     bottomSheetRef.current.close();
    // createPoolMutation.mutate(data, {
    //   onSuccess: (response) => {
    //     console.log(response);
    //     bottomSheetRef.current.close();
    //   },
    // });
  };
```

https://user-images.githubusercontent.com/1065299/198856785-de7846ad-059f-42e3-a774-54fc5ddaec8d.mp4

**Comment by iArchin** [comment]
I have the same problem, anything?

**Comment by Simoon-F** [comment]
same.

**Reply by jonacds** [reply]
same

**Comment by ri-yarm** [comment]
how did you solve this problem?

**Reply by Tjax33** [reply]
Same here. This only works with wrappingthe close function in a set timeout of 300 milliseconds.

Reason the keyboard has to be competely gone before bottom sheet will close
